tests/test_search.py

```diff
@@ -489,3 +492,35 @@ def test_search_settings_from_root(capsys, static_mixer_settings_session):
         "conduction", search_root=solver.setup.boundary_conditions.wall["wall"].phase
     )
     assert '<search_root>["<name>"].shell_conduction["<name>"] (Object)' in results
+
+
+def test_search_whole_word_parent_child(monkeypatch, capsys):
+    api_tree_data = {
+        "api_objects": [
+            "<solver_session>.parent (Object)",
+            "<solver_session>.parent.child (Parameter)",
+        ],
+        "api_tui_objects": [],
+        "all_api_object_name_synsets": {  # why doesn't _search_whole_word() work without synsets?
+            "parent": ["parent"],
+            "child": ["child"],
+        },
+    }
+    search_module = sys.modules["ansys.fluent.core.search"]
+    monkeypatch.setattr(search_module, "_get_api_tree_data", lambda: api_tree_data)
+    # _search_whole_word() should return a list. Print should happen in the top-level search() function.
```

Comment:
We have added a module level variable by using that we can either print or return results as per the need. Thank you.